cmake/HandleEigen.cmake

```diff
@@ -14,10 +14,13 @@ endif()
 
 # Switch for using system Eigen or GTSAM-bundled Eigen
 if(GTSAM_USE_SYSTEM_EIGEN)
-    find_package(Eigen3 REQUIRED) # need to find again as REQUIRED
+    find_package(Eigen3 CONFIG REQUIRED) # need to find again as REQUIRED
 
-    # Use generic Eigen include paths e.g. <Eigen/Core>
-    set(GTSAM_EIGEN_INCLUDE_FOR_INSTALL "${EIGEN3_INCLUDE_DIR}")
+    # The actual include directory (for BUILD cmake target interface):
+    # Node Eigen cmake is completelly broken on some version,
+    # EIGEN3_INCLUDE_DIR is just unusable as it points to some random location.
```

Comment:
Does this mean there is no guarantee this will work everywhere and for everyone?

Reply:
This means it is really ugly, and should not be necessary on some versions. This works since eigen 3.3.0. Prior Eigen will not, but is not supported anyway AFAIK.

Reply:
Then we should probably replace the text with a comment stating that it will work for Eigen 3.3.0 onwards but may not for prior versions. Basically what you said in your comment. 🙂

Reply:
This is also in the commit message, but I'll update the comment.

Reply:
done